src/views/lighttable.c

```diff
@@ -315,6 +328,9 @@ static void check_layout(dt_view_t *self)
   // layout has changed, let restore panels
   dt_ui_restore_panels(darktable.gui->ui);
 
+  // make sure we reset expose layout
+  _expose_destroy_slots(self);
```

Comment:
check_layout() is called from expose(), so slots are destroyed on each expose(), not sure if this is necessary or can be avoided.

Reply:
just before, check_layout return if there's no change, so this is called only on layout change.

Reply:
Didn't see that, all OK then.